### PR TITLE
Added buildspec file used by CodeBuild

### PR DESCRIPTION
**Pull request by adam-aws**
Issue #41 

Description of changes:
Setup CI for github project

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: 0.2
+
+phases:
+  install:
+    runtime-versions:
+      java: corretto11
```
Comment (robin-aws) [inline buildspec.yml:6]: This should be corretto8 to make sure we can still build/run on Java 8

File: buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pre_build:
+    commands:
+      - echo Nothing to do in the pre_build phase...
+  build:
+    commands:
+      - echo Build started on `date`
+      - mvn clean install
+      - echo Running unit tests
+      - mvn clean test
```
Comment (robin-aws) [inline buildspec.yml:15]: This isn't necessary since Travis is already running the unit tests

**Comment by adam-aws** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Java-Temporary-Client-CI
* Commit ID: 137d55ac5338522c5fbfc9a419cb31ef7171116e
* Result: SUCCEEDED
* [Build Logs](https://yutr34fq3a.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e6026554-1c70-4e23-833a-5782fe634358%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: buildspec.yml (hunk 1 above)
Comment (adam-aws) [reply buildspec.yml]: Made the change

File: buildspec.yml (hunk 2 above)
Comment (adam-aws) [reply buildspec.yml]: Removed unit tests

File: buildspec.yml (hunk 1 above)
Comment (adam-aws) [reply buildspec.yml]: Hooked up to github-codebuild-logs, getting a SignatureDoesNotMatch error, but apparently according to this https://github.com/jlhood/github-codebuild-logs/issues/15 the issue gets resolved by itself after a few days

**Comment by adam-aws** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Java-Temporary-Client-CI
* Commit ID: cc36650d2f69a3b8c0a5cd87fb07196f230b6633
* Result: FAILED
* [Build Logs](https://yutr34fq3a.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5aa714fa-6590-4bcb-b5d0-b52eb77e8fef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by adam-aws** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Java-Temporary-Client-CI
* Commit ID: cc36650d2f69a3b8c0a5cd87fb07196f230b6633
* Result: FAILED
* [Build Logs](https://yutr34fq3a.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d20db8eb-b89e-4663-b490-640ef5c87f0a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**